### ISSUE TITLE
EMDECS Interface: Transfer Inventory Receiving into EMDECS AP

### ISSUE THREAD

**Issue by giovanicani** [GCSO2/D3S-INVENTORY#124]
Once a receiving is finalized, invoice of the good received, there needs to be a procedure that will copy that inventory receiving "document" into AP (screen below) so that it can be included in the cheque run.

![image](https://cloud.githubusercontent.com/assets/5666297/15116179/4abd6d12-15d0-11e6-9b2e-e87860f448e6.png)

## Code Requirements:

- It should be a procedure;
- Parameters: IN: p_invrec_id, IN: p_use_id, OUT: ???TBD???, OUT: ???TBD???;
- Commit / Rollback will be done by the application side - not on the db procedure;
- Have as many EXCEPTION handling as possible, use the RAISE_APPLICATION_ERROR function to bring awareness of the error to the application level. Create your own customized exceptions using the type EXCEPTION. I can show you some examples once you have done some of the coding;
- Use DB link @emdecs_test / @emdecs_live to do the testing
- All DB objects will be created / updated on the TCS / D3S database environment.
- Well documented :).

**Comment by gchan-trailcon** [comment]
Giovani,
I believe we also need to keep the inventory on-hand in sync.  So I think we will need to populate the EMDECS receiving screen.  When completing a receiving, the on-hand will be updated.  And user will go to that screen to start the invoicing process as before.

**Comment by giovanicani** [comment]
correct, we will need to keep it in sync. Need to think how we can make this happen since all issuing, adjustment, transfers will take place in D3S.

Perhaps we need to do a batch job overnight to adjust EMDECS.

**Comment by giovanicani** [comment]
@gchan-trailcon can you please provide update. Thanks

**Comment by gchan-trailcon** [comment]
I think we need to sync the inventory data in real time from D3S to EMDECS, instead of overnight.  This way, AP invoices can be generated.  This will be a big job.  I have not started on this.
